sapi/fpm/fpm/fpm_unix.c

```diff
@@ -63,6 +64,25 @@ static struct passwd *fpm_unix_get_passwd(struct fpm_worker_pool_s *wp, const ch
 	return pwd;
 }
 
+static inline bool fpm_unix_check_listen_address(struct fpm_worker_pool_s *wp, const char *address, int flags)
+{
+    struct sockaddr_un test_socket;
+    size_t address_length = strlen(address);
+    size_t socket_length = sizeof(test_socket.sun_path);
+
+    if (wp->listen_address_domain != FPM_AF_UNIX) {
+        return true;
+    }
+
+    if (address_length < socket_length) {
+        return true;
+    }
+
+    zlog(flags, "[pool %s] cannot bind to UNIX socket '%s' as path is too long (found length: %lu, maximal length: %lu)", wp->config->name, address, address_length, socket_length);
+
+    return false;
```

Comment:
Just a note that this sort of makes it slightly inconsistent as the test fails but the actual run might still work trimmed. I guess it probably makes sense to fail the test as most of the time this is not what users would want.

There should be a special test for that.

Reply:
Yes, I saw that inconsistency as well. I tried to mention, that in my PR description "I decided it to be a warning, because you can still run against the newly created socket file but fail it for the configuration test. It felt right to do it like that."

Reply:
Ok will leave this open as we might need a note in UPGRADING for this which I will add before merging.